### PR TITLE
Adds Dates support for FsDataProviderExport (based on #258)

### PR DESCRIPTION
**Pull request by zbraniecki**
Based on #258 this adds support for exporting dates key in FsDataProviderExport.

### CODE REVIEW

**Comment by zbraniecki** [comment]
@sffc I tried to produce the result using this PR:

```
▶ cargo run --features export-bin -- --cldr-core ~/projects/intl-measurements/icu4x/data/cldr/cldr-core --cldr-dates ~/projects/intl-measurements/icu4x/data/cldr/cldr-dates-modern --all-keys --out ~/projects/intl-measurements/icu4x/data/icu4x/json2
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `/Users/zbraniecki/projects/icu4x/target/debug/icu4x-cldr-export --cldr-core /Users/zbraniecki/projects/intl-measurements/icu4x/data/cldr/cldr-core --cldr-dates /Users/zbraniecki/projects/intl-measurements/icu4x/data/cldr/cldr-dates-modern --all-keys --out /Users/zbraniecki/projects/intl-measurements/icu4x/data/icu4x/json2`
Error: Unsupported data key: dates/gregory@1
```

Can you help me debug it? What am I missing?

**Comment by zbraniecki** [comment]
The error has been fixed in the previous PR. This PR is ready for review!

**Review by sffc — DISMISSED** [review]
LGTM with the understanding that most of this PR is already covered by #258!

File: components/cldr-json-data-provider/src/transform/mod.rs [hunk 1]
```diff
@@ -10,22 +12,28 @@ use icu_data_provider::prelude::*;
...
-        self.plurals
-            .try_load(req, &self.cldr_paths)?
-            .ok_or(DataError::UnsupportedDataKey(req.data_key))
+        if let Some(result) = self.plurals.try_load(req, &self.cldr_paths)? {
+            return Ok(result);
+        }
+        if let Some(result) = self.dates.try_load(req, &self.cldr_paths)? {
+            return Ok(result);
+        }
+        Err(DataError::UnsupportedDataKey(req.data_key))
```
[the remark this answers is not in the document]
Comment (zbraniecki) [reply components/cldr-json-data-provider/src/transform/mod.rs:36]: Rebased on top of the current #258.

**Review by sffc — APPROVED** [review]

**Pull request merged**